BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListProcessor.swift

```diff
@@ -354,21 +490,68 @@ extension VaultAutofillListProcessor {
         }
     }
 
-    /// Picks a cipher to use for the Fido2 creation process
+    func loadFido2Section(
+        searchText: String? = nil,
+        searchResults: [CipherView]? = nil
+    ) async throws -> VaultListSection? {
+        guard let fido2Credentials = services.fido2UserInterfaceHelper.availableCredentialsForAuthentication,
+              !fido2Credentials.isEmpty,
+              let fido2Delegate = appExtensionDelegate as? Fido2AppExtensionDelegate,
+              case let .autofillFido2VaultList(_, parameters) = fido2Delegate.extensionMode else {
+            return nil
+        }
+
+        var filteredFido2Credentials = fido2Credentials
+        if let searchResults {
+            filteredFido2Credentials = filteredFido2Credentials.filter { cipher in
+                searchResults.contains(where: { $0.id == cipher.id })
+            }
+        }
+
+        guard !filteredFido2Credentials.isEmpty else {
+            return nil
+        }
+
+        let fido2ListItems = try await filteredFido2Credentials
+            .asyncMap { cipher in
+                let fido2CredentialAutofillView = try await self.services.clientService
+                    .platform()
+                    .fido2()
+                    .decryptFido2AutofillCredentials(cipherView: cipher)
+
+                return VaultListItem(
+                    cipherView: cipher,
+                    fido2CredentialAutofillView: fido2CredentialAutofillView[0]
```

Comment:
❓ Is there any risk that `fido2CredentialAutofillView` would be an empty array? Any reason to not add a guard for getting `fido2CredentialAutofillView.first` instead?